functions/Invoke-DbaDbccFreeCache.ps1

```diff
@@ -0,0 +1,191 @@
+
+    .NOTES
+        Tags: DBCC
+        Author: Patrick Flynn (@sqllensman)
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Invoke-DbaDbccFreeCache
+
+    .EXAMPLE
+        PS C:\> Invoke-DbaDbccFreeCache -SqlInstance SqlServer2017 -Operation FREEPROCCACHE
+
+        Runs the command DBCC FREEPROCCACHE against the instance SqlServer2017 using Windows Authentication
+
+    .EXAMPLE
+        PS C:\> Invoke-DbaDbccFreeCache -SqlInstance SqlServer2017 -Operation FREESESSIONCACHE -NoInformationalMessages
+
+        Runs the command DBCC FREESESSIONCACHE WITH NO_INFOMSGS against the instance SqlServer2017 using Windows Authentication
+
+    .EXAMPLE
+        PS C:\> Invoke-DbaDbccFreeCache -SqlInstance SqlServer2017 -Operation FREESYSTEMCACHE -NoInformationalMessages
+
+        Runs the command DBCC FREESYSTEMCACHE WITH NO_INFOMSGS against the instance SqlServer2017 using Windows Authentication
+
+    .EXAMPLE
+        PS C:\> Invoke-DbaDbccFreeCache -SqlInstance SqlServer2017 -Operation FREEPROCCACHE -DBCCInputValue 0x060006001ECA270EC0215D05000000000000000000000000
+
+        Remove a specific plan with plan_handle 0x060006001ECA270EC0215D05000000000000000000000000 from the cache via the command DBCC FREEPROCCACHE(0x060006001ECA270EC0215D05000000000000000000000000) against the instance SqlServer2017 using Windows Authentication
+
+    .EXAMPLE
+        PS C:\> Invoke-DbaDbccFreeCache -SqlInstance SqlServer2017 -Operation FREEPROCCACHE -DBCCInputValue default
+
+        Runs the command DBCC FREEPROCCACHE('default') against the instance SqlServer2017 using Windows Authentication. This clears all cache entries associated with a resource pool 'default'.
+
+    .EXAMPLE
+        PS C:\> Invoke-DbaDbccFreeCache -SqlInstance SqlServer2017 -Operation FREESYSTEMCACHE -DBCCInputValue default
+
+        Runs the command DBCC FREESYSTEMCACHE ('ALL', default) against the instance SqlServer2017 using Windows Authentication. This will clean all the caches with entries specific to the resource pool named "default".
+
+    .EXAMPLE
+        PS C:\> Invoke-DbaDbccFreeCache -SqlInstance SqlServer2017 -Operation FREESYSTEMCACHE -DBCCInputValue default -MarkInUseForRemoval
+
+        Runs the command DBCC FREESYSTEMCACHE ('ALL', default) WITH MARK_IN_USE_FOR_REMOVAL against the instance SqlServer2017 using Windows Authentication. This will to release entries once the entries become unused for all the caches with entries specific to the resource pool named "default".
+
+    #>
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [ValidateSet('FREEPROCCACHE', 'FREESESSIONCACHE', 'FREESYSTEMCACHE')]
```

Comment:
I'm on the fence, what do you think about `FreeProcCache` and `FreeSessionCache` or are we all just so used to seeing all caps that we could let this go and be in the style of DBCC? (Tho we did go with Dbcc so I'm leaning FreeProcCache)

Reply:
We can be consistent and go with FreeProcCache. This was first command and I was going with what usef to seeing. Also do we want to change DBCCInputValue to InputValue?

Reply:
oh, yes that would be good too